base/set.jl

```diff
@@ -17,6 +17,37 @@ Set() = Set{Any}()
 Construct a [`Set`](@ref) of the values generated by the given iterable object, or an
 empty set. Should be used instead of [`BitSet`](@ref) for sparse integer sets, or
 for sets of arbitrary objects.
+
+# Examples
+```jldoctest
+julia> animals = Set{String}(["cat","fish","dog","bird"])
+Set(["fish", "cat", "bird", "dog"])
+```
+
+You can let Julia infer the type:
```

Comment:
I think sentences like this are better written like

> If the element type is omitted it is computed by intersecting the types of the arguments.

or something like that.